packages/medusa/src/api/routes/admin/customers/update-customer.ts

```diff
@@ -94,4 +121,8 @@ export class AdminPostCustomersCustomerReq {
   @IsObject()
   @IsOptional()
   metadata?: object
+
+  @IsArray()
+  @IsOptional()
+  groups?: Group[]
```

Comment:
The `class-validator` doesn't automatically validate the nested classes (i.e. `Group`) unless we use the decorator `ValidateNested`.

```suggestion

  @ValidateNested()
  @IsArray()
  @IsOptional()
  groups?: Group[]
```